ui/dashboards/src/components/GridLayout/GridLayout.tsx

```diff
@@ -49,6 +52,18 @@ export function GridLayout(props: GridLayoutProps) {
     }
   };
 
+  const handleWidthChange = (
+    containerWidth: number,
+    margin: [number, number],
+    cols: number,
+    containerPadding: [number, number]
+  ) => {
+    const marginX = margin[0];
+    const marginWidth = marginX * (cols - 1); // exclude margin from container width
+    const containerPaddingWidth = containerPadding[0] * 2;
+    setGridColWidth(Math.round((containerWidth - marginWidth - containerPaddingWidth) / cols));
```

Comment:
```suggestion
    setGridColWidth((containerWidth - marginWidth - containerPaddingWidth) / cols);
```

Looks like the math is still not quite right. If you remove the rounding here and put it in the other function, then the numbers match up properly.

Reply:
Interesting, when i tested, it was more accurate when I moved the rounding here instead of the other function. But i can move it back. I don't think the suggestedStepMs needs to be exact

Reply:
Ah weird. It was several pixels off with the other way around, but perfectly matched when I did it this way.

Reply:
I wish we could write an automated test for this, but I think that's kind a pain with how this code is written right now, and I don't want to add more overhead to this already sufficiently complex PR.